**Code for understanding this file:**
- <font color=blue>Questions are put in blue <font color=black> 
- Answers are put in black

# Probability practice

## Part (a):

<font color=blue>
Visitors to your website are asked to answer a single survey question before they get access to the content on the page. Among all of the users, there are two categories: Random Clicker (RC), and Truthful Clicker (TC). There are two possible answers to the survey: yes and no. Random clickers would click either one with equal probability. You are also giving the information that the expected fraction of random clickers is 0.3. After a trial period, you get the following survey results: 65% said Yes and 35% said No. What fraction of people who are truthful clickers answered yes? Hint: use the rule of total probability.

- P(YES|RC) = 0.50
- P(NO|RC) = 0.50
- P(RC) = 0.30
- P(YES) = 0.65
- P(NO) = 0.35

Trying to find: P(Y|TC) (*the probability of people who answered yes **given** they are truthful clickers.


$Total Probability Rule: P(TC = Y, RC = Y), + P(TC = Y, RC = N) + P(TC = N, RC = Y)+ P(TC = N, RC = N) $

$P(YES) = P(YES | RC) * P(RC) + P(YES | TC) * P(TC) $

$0.65 = (0.5 * 0.3) + P(YES|TC) * 0.70$

$P(YES|TC) = (0.65 - (0.5*0.3))/0.70$

In [1]:
prob_yes_tc = (0.65 - (0.5*0.3))/0.70
print("P(YES|TC):", prob_yes_tc)

P(YES|TC): 0.7142857142857143


The fraction of people who are truthful clickers answered yes is 0.7143, or 71.43%.

## Part (b):

<font color=blue>

Imagine a medical test for a disease with the following two attributes:

- The sensitivity is about 0.993. That is, if someone has the disease, there is a probability of 0.993 that they will test positive.
- The specificity is about 0.9999. This means that if someone doesn't have the disease, there is probability of 0.9999 that they will test negative.
- In the general population, incidence of the disease is reasonably rare: about 0.0025% of all people have it (or 0.000025 as a decimal probability).

Suppose someone tests positive. What is the probability that they have the disease?

- Senstitivty = P(POS|DISEASED) = 0.993
- From here, can calculate the probability of being neg, given diseased: P(NEG|DISEASED) = 0.007
- Specificity = P(NEG|NOT DISEASED) = 0.9999
- From here, can calculate the probability of being positive, given not diseased: P(POS|NOT DISEASED) = 0.0001 <br><br>
    
    
- P(DISEASED) = 0.000025
- P(NOT DISEASED) = 0.999975

Trying to find: P(DISEASED|POS)

I can use **Bayes Theorem** to find this answer. First, I need to find the probability of being diseased:

$P(POS) = P(POS|DISEASED) * P(DISEASED) + P(POS|NOT DISEASED) * P(NOT DISEASED)$


$P(POS) = (0.993 * 0.000025) + (0.0001 * 0.999975)$

In [2]:
prob_positive = (0.993 * 0.000025) + (0.0001 * 0.999975)
print("P(POS):", prob_positive)

P(POS): 0.0001248225


$P(DISEASED|POS) = P(POS|DISEASED) * P(DISEASED) / P(POSITIVE)$

$P(DISEASED|POS) = 0.993 * 0.000025 / 0.0001248225 $

In [3]:
prob_diseased_pos = 0.993 * 0.000025 / 0.0001248225
print("P(DISEASED|POS):", prob_diseased_pos)

P(DISEASED|POS): 0.19888241302649762


The probability that a person has the disease, given that they have tested positive is 0.19888 as a fraction.